src/server/rdb_save.cc

```diff
@@ -1032,7 +1032,7 @@ error_code RdbSaver::Impl::ConsumeChannel(const Cancellation* cll) {
     pushed_bytes += ptr->channel_bytes();
   }
 
-  DCHECK(!channel_.TryPop(*record));
```

Comment:
in what case you saw this failing?

Reply:
I think it invoked undefined behavior already before, but apparently it didn't usually crash?

If you add `DCHECK(record.has_value())` and run the following test case -

```
./dragonfly --port 6000&
export PID1=$!
./dragonfly --port 6001&\
export PID2=$!
sleep 1 && redis-cli -p 6001 replicaof localhost 6000
sleep 10 && kill $PID1 $PID2 # Cleanup
```

It hits the assert immediately